Assets/Scripts/Models/Utility.cs

```diff
@@ -0,0 +1,802 @@
+#region License
+// ====================================================
+// Project Porcupine Copyright(C) 2016 Team Porcupine
+// This program comes with ABSOLUTELY NO WARRANTY; This is free software, 
+// and you are welcome to redistribute it under certain conditions; See 
+// file LICENSE, which is part of this source code package, for details.
+// ====================================================
+#endregion
+using System;
+using System.Collections.Generic;
+using System.Xml;
+using System.Xml.Schema;
+using System.Xml.Serialization;
+using Animation;
+using MoonSharp.Interpreter;
+using MoonSharp.Interpreter.Interop;
+using ProjectPorcupine.PowerNetwork;
+using UnityEngine;
+
+/// <summary>
+/// InstalledObjects are things like walls, doors, and utility (e.g. a sofa).
+/// </summary>
+[MoonSharpUserData]
+public class Utility : IXmlSerializable, ISelectable, IContextActionProvider, IBuildable
+{
+    // Prevent construction too close to the world's edge
+    private const int MinEdgeDistance = 5;
+
+    /// <summary>
+    /// This action is called to get the sprite name based on the utility parameters.
+    /// </summary>
+    private string getSpriteNameAction;
+
+    /// <summary>
+    /// These context menu lua action are used to build the context menu of the utility.
+    /// </summary>
+    private List<ContextMenuLuaAction> contextMenuLuaActions;
+
+    /// <summary>
+    /// Custom parameter for this particular piece of utility.  We are
+    /// using a custom Parameter class because later, custom LUA function will be
+    /// able to use whatever parameters the user/modder would like, and contain strings or floats.
+    /// Basically, the LUA code will bind to this Parameter.
+    /// </summary>
+    private Parameter utilParameters;
+
+    private List<Job> jobs;
```

Comment:
I would assume repair jobs would be created by the utility, once we have breaking and repairing stuff in.